Sources/SendGridKit/Models/WebhookEvent.swift

```diff
@@ -0,0 +1,889 @@
+import Foundation
+
+/// Main webhook event enum that encompasses all SendGrid webhook event types
+public enum SendGridWebhookEvent: Codable, Sendable {
+    case delivery(SendGridDeliveryEvent)
+    case engagement(SendGridEngagementEvent)
+    case accountStatusChange(SendGridAccountStatusChangeEvent)
+    case received(SendGridReceivedEvent)
+
+    /// Common event types used for initial parsing
+    public enum EventType: String, Codable, Sendable {
+        case bounce
+        case click
+        case deferred
+        case delivered
+        case dropped
+        case groupResubscribe = "group_resubscribe"
+        case groupUnsubscribe = "group_unsubscribe"
+        case open
+        case processed
+        case spamreport
+        case unsubscribe
+        case blocked
+        case accountStatusChange = "account_status_change"
+        case received
+    }
+
+    /// Shared structures used across different event types
+    public struct Pool: Codable, Sendable {
+        public let name: String
+        public let id: Int
+    }
+
+    public struct Newsletter: Codable, Sendable {
+        public let newsletterUserListId: String
+        public let newsletterId: String
+        public let newsletterSendId: String
+
+        enum CodingKeys: String, CodingKey {
+            case newsletterUserListId = "newsletter_user_list_id"
+            case newsletterId = "newsletter_id"
+            case newsletterSendId = "newsletter_send_id"
+        }
+    }
```

Comment:
A minor stylistic issue: in this package, when the term ID appears together with other words in a property, we always capitalise both letters.
For example:
```suggestion
    public struct Newsletter: Codable, Sendable {
        public let newsletterUserListID: String
        public let newsletterID: String
        public let newsletterSendID: String

        enum CodingKeys: String, CodingKey {
            case newsletterUserListID = "newsletter_user_list_id"
            case newsletterID = "newsletter_id"
            case newsletterSendID = "newsletter_send_id"
        }
    }
```